shared/units/units.go

```diff
@@ -5,6 +5,19 @@ import (
 	"strconv"
 )
 
+func handleOverflow(val int64, mult int64) (int64, error) {
+	result := val * mult
+	if val == 0 || mult == 0 || val == 1 || mult == 1 {
+		return result, nil
+	}
+
+	if val != 0 && (result/val) != mult {
+		return 0, fmt.Errorf("Overflow multiplying %d with %d", val, mult)
```

Comment:
other error conditions are returning -1, error, why this would return 0, error?

Reply:
Good catch.